libvips/foreign/dzsave.c

```diff
@@ -326,6 +326,14 @@ typedef struct _VipsGsfDirectory {
 	 */
         GsfOutput *container;
 
+	/* Track number of files for ZIP64 limit check.
+	 */
+	unsigned int file_count;
```

Comment:
I think I'd just use an int. Mixed uint/int arithmetic is very error-prone, and there's no need for the extra range a uint would give you.